userspace/engine/rule_loader.cpp

```diff
@@ -36,34 +36,80 @@ using namespace std;
 using namespace libsinsp::filter;
 
 rule_loader::context::context(const std::string& name)
-	: name(name)
 {
+	location loc = {name, YAML::Mark(), "rules content", ""};
+	m_locs.push_back(loc);
 }
 
 rule_loader::context::context(const YAML::Node &item,
 			      const std::string item_type,
 			      const std::string item_name,
 			      const context& parent)
-	: name(parent.name)
+{
+	init(parent.name(), item.Mark(), item_type, item_name, parent);
+}
+
+rule_loader::context::context(const libsinsp::filter::parser::pos_info& pos,
+			      const std::string& condition,
+			      const context& parent)
+	: alt_content(condition)
+{
+
+	// Contexts based on conditions don't use the
+	// filename. Instead the "name" is just the condition, and
+	// uses a short prefix of the condition.
+	std::string name = "\"" + condition.substr(0, 20) + "...\"";
+	std::replace(name.begin(), name.end(), '\n', ' ');
+	std::replace(name.begin(), name.end(), '\r', ' ');
+
+	std::string item_type = "condition expression";
```

Comment:
We're starting to have many of these types. Can't we have an enum or something to centralize them?

Reply:
Yes, I made an enum.